src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -340,6 +342,18 @@ def render_definition_template(
         project_context[CONTEXT_KEY]["env"] = environment_overrides
         return ProjectProperties(project_definition, project_context)
 
+    has_user_referenced_vars = (
+        len(_get_referenced_vars_in_definition(template_env, definition)) > 0
```

Comment:
is there a risk/side effect from calling this?

Reply:
I went through the called functions and i do not believe there should be any, unless `self._jinja_env.parse` has side effects?

Reply:
it could throw error, but I guess it will throw them later anyways